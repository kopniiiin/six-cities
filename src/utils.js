const MONTHS = [
  `January`,
  `February`,
  `March`,
  `April`,
  `May`,
  `June`,
  `July`,
  `August`,
  `September`,
  `October`,
  `November`,
  `December`
];

export const extend = (target, update) => Object.assign({}, target, update);

export const pushElement = (array, element) => [...array, element];

export const replaceElement = (array, element, index) => [...array.slice(0, index), element, ...array.slice(index + 1)];

export const removeElement = (array, index) => [...array.slice(0, index), ...array.slice(index + 1)];

export const upperCaseFirstLetter = (string) => string[0].toUpperCase() + string.slice(1);

export const formatDate = (dateString) => {
  const date = new Date(dateString);
  return `${MONTHS[date.getMonth()]} ${date.getDate()}, ${date.getFullYear()}`;
};
