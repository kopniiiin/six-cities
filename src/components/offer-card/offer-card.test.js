import React from "react";
import {shallow} from "enzyme";

import {extend} from "../../utils";

import OfferCard from "./offer-card";

import testMocks from "../../test-mocks/offer-card";

describe(`snapshot test: OfferCard component`, () => {
  it(`should render correctly`, () => expect(
      shallow(<OfferCard {...testMocks}/>)
  ).toMatchSnapshot());

  it(`should render premium mark`, () => expect(
      shallow(<OfferCard {...extend(testMocks, {isPremium: true})}/>)
  ).toMatchSnapshot());
});
