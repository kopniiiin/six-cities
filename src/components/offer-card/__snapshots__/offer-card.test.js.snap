// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test: OfferCard component should render correctly 1`] = `
<article
  className="block__place-card place-card"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="block__image-wrapper place-card__image-wrapper"
  >
    <a
      href="#"
    >
      <img
        alt="Place photo"
        className="place-card__image"
        height="200"
        src="photo"
        width="260"
      />
    </a>
  </div>
  <div
    className="place-card__info"
  >
    <div
      className="place-card__price-wrapper"
    >
      <div
        className="place-card__price"
      >
        <b
          className="place-card__price-value"
        >
          €
          4
        </b>
        <span
          className="place-card__price-text"
        >
          / night
        </span>
      </div>
      <BookmarkButton
        blockClassName="place-card"
        isActive={false}
        isBig={false}
      />
    </div>
    <StarRating
      blockClassName="place-card"
      isValueShown={false}
      value={4}
    />
    <h2
      className="place-card__name"
    >
      <a
        href="#"
        onClick={[Function]}
      >
        Paper place
      </a>
    </h2>
    <p
      className="place-card__type"
    >
      Room
    </p>
  </div>
</article>
`;

exports[`snapshot test: OfferCard component should render premium mark 1`] = `
<article
  className="block__place-card place-card"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <PremiumMark
    blockClassName="place-card"
  />
  <div
    className="block__image-wrapper place-card__image-wrapper"
  >
    <a
      href="#"
    >
      <img
        alt="Place photo"
        className="place-card__image"
        height="200"
        src="photo"
        width="260"
      />
    </a>
  </div>
  <div
    className="place-card__info"
  >
    <div
      className="place-card__price-wrapper"
    >
      <div
        className="place-card__price"
      >
        <b
          className="place-card__price-value"
        >
          €
          4
        </b>
        <span
          className="place-card__price-text"
        >
          / night
        </span>
      </div>
      <BookmarkButton
        blockClassName="place-card"
        isActive={false}
        isBig={false}
      />
    </div>
    <StarRating
      blockClassName="place-card"
      isValueShown={false}
      value={4}
    />
    <h2
      className="place-card__name"
    >
      <a
        href="#"
        onClick={[Function]}
      >
        Paper place
      </a>
    </h2>
    <p
      className="place-card__type"
    >
      Room
    </p>
  </div>
</article>
`;
